cmd/otelcorecol/builder-config.yaml

```diff
@@ -1,3 +1,5 @@
+SkipCompilation: true
```

Comment:
@jpkrohling this is possible right now, but I think would be good to actually define a `mapstructure` tag with a value like `skip_compilation`? What do you think?

Reply:
What advantage does this have over a CLI flag?  It seems that this would prevent use of this config to directly build a collector binary.  It may not be used that way by `make genotelcorecol`, but why foreclose the option?

Reply:
I am saying that right now, the way how viper works, makes this available. Is this what we want? Is this the name we want?

Reply:
Question: did you know that this is possible? 

You do have a good point on keeping the flag, but do we need this config then?

Reply:
> Is this what we want? Is this the name we want?

I wasn't aware and am surprised that it works, although it does make sense now that you pointed it out. I see two different config types: runtime (CLI flags) and distribution-related config. They are internally stored in the same struct, but are logically different.

Reply:
@jpkrohling can you take this and make a proposal to fix this one way or the other?

Reply:
I would just not document this flag and not support it being used this way. Not sure we need a way to actively prevent people from using it.

Reply:
If you want to avoid the config support you can have a "mapstructure:-"

Reply:
True, that's an easy fix.

Reply:
@jpkrohling you are not moving away from owning this :))

Reply:
lol , that wasn't my intention :-) I'll change this PR tomorrow morning.

Reply:
@jpkrohling it looks like someone is keep trying to escape from this 🗡️

Reply:
I do have this in my queue but didn't find the time yet to do the coding... Too many code reviews to perform :-)

Reply:
Maybe you can split the PRs to reduce your review time 😜

Reply:
Touche!

Reply:
Done!